### PR TITLE
Add sqlServerSchema command in data migration

### PR DESCRIPTION
**Pull request by yadongyaly**
The PR is to add new CLI commands "az datamigration sql-server-schema". The same command for Powershell has been published. The new CLI commands has passed local tests.

The local test commands:
az datamigration sql-server-schema --action "MigrateSchema" --src-sql-connection-str "Server=;Initial Catalog=;User ID=;Password=" --tgt-sql-connection-str "Server=;Initial Catalog=;User ID=;Password="
az datamigration sql-server-schema --action "GenerateScript" --src-sql-connection-str "Server=;Initial Catalog=;User ID=;Password=" --tgt-sql-connection-str "Server=;Initial Catalog=;User ID=;Password=" --output-folder "C:\OutputFolder"
az datamigration sql-server-schema --action "GenerateScript" --src-sql-connection-str "Server=;Initial Catalog=;User ID=;Password=" --tgt-sql-connection-str "Server=;Initial Catalog=;User ID=;Password=" --input-script-file-path "C:\OutputFolder\script.sql"
az datamigration sql-server-schema --config-file-path "C:\configfile.json"

I am from the service team.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Review by siyao-Siyang — APPROVED** [review]

File: src/datamigration/azext_datamigration/manual/custom.py [hunk 1]
```diff
@@ -341,3 +341,83 @@ def datamigration_tde_migration(source_sql_connection_string=None,
...
+# -----------------------------------------------------------------------------------------------------------------
+def datamigration_sql_server_schema(action=None,
+                                    src_sql_connection_str=None,
+                                    tgt_sql_connection_str=None,
+                                    input_script_file_path=None,
+                                    output_folder=None,
+                                    config_file_path=None):
+
+    # Error message strings
+    actionError = "Please provide valid action value. The valid values are: MigrateSchema, GenerateScript, DeploySchema."
```
Comment (evelyn-ys) [inline src/datamigration/azext_datamigration/manual/custom.py:357]: Then you can omit the check here

**Review by evelyn-ys — COMMENTED** [review]
Please add tests for new command

**Comment by yadongyaly** [comment]
> Please add tests for new command

Thanks for reviewing @evelyn-ys , Is my local test good enough? 

I did local CLI test for the new command. Below are test results ( I removed the actual value of user name and password)
Test 1: az datamigration sql-server-schema --action "MigrateSchema" --src-sql-connection-str "Server=;Initial Catalog=;User ID=;Password=" --tgt-sql-connection-str "Server=;Initial Catalog=;User ID=;Password="

Result :
Starting schema migration...

Total number of object collected: 35
Total elapsed time: 00:00:06.9587061
DDL triggers: 4
Full text catalogs: 2
Plan guides: 2
Roles: 1
Schemas: 5
Stored procedures: 5
Tables: 10
User defined functions: 4
Users: 2

The number of scripted objects: 55
Progress: 100%
Total elapsed time: 00:00:39.5761905

The number of deployed script batches: 110/110
Progress: 100%
Total elapsed time: 00:00:06.0982252


The event and error logs folder path: C:\Users\XXX\AppData\Local\Microsoft\SqlSchemaMigration
Finishing schema migration.


Test 2: az datamigration sql-server-schema --action "GenerateScript" --src-sql-connection-str "Server=;Initial Catalog=;User ID=;Password=" --tgt-sql-connection-str "Server=;Initial Catalog=;User ID=;Password=" --output-folder "C:\OutputFolder"



The generated TSQL schema script folder path: C:\OutputFolder\SchemaScript-20230606.sql
The event and error logs folder path: C:\Users\XXX\AppData\Local\Microsoft\SqlSchemaMigration
Finishing schema script generation.


Test 3: az datamigration sql-server-schema --action "DeploySchema" --src-sql-connection-str "Server=;Initial Catalog=;User ID=;Password=" --tgt-sql-connection-str "Server=;Initial Catalog=;User ID=;Password=" --input-script-file-path "C:\OutputFolder\script.sql"

Result:
Starting schema deployment...

The number of deployed script batches: 110/110
Progress: 100%
Total elapsed time: 00:00:05.6998954


The event and error logs folder path: C:\Users\XXX\AppData\Local\Microsoft\SqlSchemaMigration
Finishing schema deployment.



Test 4: az datamigration sql-server-schema --config-file-path "C:\configfile.json"

Result:
Run through the provided config file:
Starting schema script generation...

Total number of object collected: 35
Total elapsed time: 00:00:07.3111221
DDL triggers: 4
Full text catalogs: 2
Plan guides: 2
Roles: 1
Schemas: 5
Stored procedures: 5
Tables: 10
User defined functions: 4
Users: 2

The number of scripted objects: 55
Progress: 100%
Total elapsed time: 00:00:38.2304043


The generated TSQL schema script folder path: C:\Users\yaly.REDMOND\AppData\Local\Microsoft\SqlSchemaMigration\SchemaScript-20230606.sql
The event and error logs folder path: C:\Users\XXX\AppData\Local\Microsoft\SqlSchemaMigration
Finishing schema script generation.

Test 5: az datamigration sql-server-schema --action "D2eploySchema" --src-sql-connection-str "Server=;Initial Catalog=;User ID=;Password=" --tgt-sql-connection-str "Server=;Initial Catalog=;User ID=;Password=" --output-folder "C:\OutputFolder"

Result:
az datamigration sql-server-schema: 'D2eploySchema' is not a valid value for '--action'. Allowed values: MigrateSchema, GenerateScript, DeploySchema.
Did you mean 'DeploySchema' ?

Examples from AI knowledge base:
az datamigration sql-server-schema --action "GenerateScript" --src-sql-connection-str "Server=;Initial Catalog=;User ID=;Password=" --tgt-sql-connection-str "Server=;Initial Catalog=;User ID=;Password=" --input-script-file-path "C:\OutputFolder\script.sql"
Run Deploy TSQL script to the target Azure SQL Database using Parameters.

az datamigration sql-server-schema --action "MigrateSchema" --src-sql-connection-str "Server=;Initial Catalog=;User ID=;Password=" --tgt-sql-connection-str "Server=;Initial Catalog=;User ID=;Password="
Run Migrate database objects from the source SQL Server to the target Azure SQL Database using Parameters.

az datamigration sql-server-schema --action "GenerateScript" --src-sql-connection-str "Server=;Initial Catalog=;User ID=;Password=" --tgt-sql-connection-str "Server=;Initial Catalog=;User ID=;Password=" --output-folder "C:\OutputFolder"
Run Generate TSQL schema script from the source SQL Server using Parameters.

https://aka.ms/cli_ref
Read more about the command in reference docs

**Review by evelyn-ys — COMMENTED** [review]
Why don't you add your local test in [azext_datamigration/manual/tests/latest/test_datamigration_scenario.py](https://github.com/Azure/azure-cli-extensions/blob/main/src/datamigration/azext_datamigration/manual/tests/latest/test_datamigration_scenario.py) or [azext_datamigration/tests/latest/test_datamigration_scenario.py](https://github.com/Azure/azure-cli-extensions/blob/main/src/datamigration/azext_datamigration/tests/latest/test_datamigration_scenario.py)?

**Comment by yadongyaly** [comment]
> Why don't you add your local test in [azext_datamigration/manual/tests/latest/test_datamigration_scenario.py](https://github.com/Azure/azure-cli-extensions/blob/main/src/datamigration/azext_datamigration/manual/tests/latest/test_datamigration_scenario.py) or [azext_datamigration/tests/latest/test_datamigration_scenario.py](https://github.com/Azure/azure-cli-extensions/blob/main/src/datamigration/azext_datamigration/tests/latest/test_datamigration_scenario.py)?

I do not see previous customized commands added the test there.  

For this new command, in my local test, my local machine needs to have connections to a source sql server and a target sql server. With the connection strings to both source and target, the command can do the migration work. 

In this case, I think it will be hard to add the integration tests in pipeline?

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**